### PR TITLE
LibWeb: Even more IDL fixes :makemore:

### PR DESCRIPTION
**Pull request by IdanHo**

### CODE REVIEW

File: Meta/Lagom/Tools/CodeGenerators/LibWeb/WrapperGenerator/IDLGenerators.cpp [hunk 1]
```diff
@@ -560,7 +560,16 @@ static void generate_to_cpp(SourceGenerator& generator, ParameterType& parameter
     } else if (interface.enumerations.contains(parameter.type->name)) {
         auto enum_generator = scoped_generator.fork();
         auto& enumeration = interface.enumerations.find(parameter.type->name)->value;
-        enum_generator.set("enum.default.cpp_value", *enumeration.translated_cpp_names.get(optional_default_value.value_or(enumeration.first_member)));
+        StringView enum_member_name;
+        if (optional_default_value.has_value()) {
+            VERIFY(optional_default_value->length() >= 3 && (*optional_default_value)[0] == '"' && (*optional_default_value)[optional_default_value->length() - 1] == '"');
```
Comment (alimpfard) [inline Meta/Lagom/Tools/CodeGenerators/LibWeb/WrapperGenerator/IDLGenerators.cpp:563]: Can't the default be `""`? XHR has a `""` enum, though it's not set as a default anywhere I know 🤔

File: Userland/Libraries/LibWeb/IntersectionObserver/IntersectionObserver.h [hunk 2]
```diff
@@ -13,9 +13,9 @@
 namespace Web::IntersectionObserver {
 
 struct IntersectionObserverInit {
-    DOM::Node* root { nullptr };
+    Optional<Variant<NonnullRefPtr<DOM::Element>, NonnullRefPtr<DOM::Document>>> root;
```
Comment (alimpfard) [inline Userland/Libraries/LibWeb/IntersectionObserver/IntersectionObserver.h:16]: :yakkie:

**Review by alimpfard — COMMENTED** [review]
Just a tiny nitpick

File: Meta/Lagom/Tools/CodeGenerators/LibWeb/WrapperGenerator/IDLGenerators.cpp (hunk 1 above)
Comment (IdanHo) [reply Meta/Lagom/Tools/CodeGenerators/LibWeb/WrapperGenerator/IDLGenerators.cpp:563]: That's a good point, I'll make it `>= 2` instead

**Review by alimpfard — APPROVED** [review]

**Pull request merged**